src/firefly/js/tables/ui/TablePanel.jsx

```diff
@@ -171,7 +186,7 @@ export class TablePanel extends PureComponent {
                                         title={TT_VIEW}
                                         className={viewIcoStyle}/>
                                 {showSave &&
-                                    <div onClick={this.saveTable}
+                                    <div onClick={showTableDownloadDialog(tableConnector)}
```

Comment:
If you don't mind, can I ask you to move the TableSave logic to a separate file?  So much was added to this file that I felt it would be cleaner if the logic for TableSave would be in TableSave.js and referenced from here.